lib/ansible/plugins/connection/ssh.py

```diff
@@ -1241,31 +1221,16 @@ def _file_transport_command(self, in_path: str, out_path: str, sftp_action: str)
         # Transfer methods to try
         methods = []
 
-        # Use the transfer_method option if set, otherwise use scp_if_ssh
+        # Use the transfer_method option if set
         ssh_transfer_method = self.get_option('ssh_transfer_method')
-        scp_if_ssh = self.get_option('scp_if_ssh')
-        if ssh_transfer_method is None and scp_if_ssh == 'smart':
+        if ssh_transfer_method is None:
             ssh_transfer_method = 'smart'
 
         if ssh_transfer_method is not None:
             if ssh_transfer_method == 'smart':
                 methods = smart_methods
             else:
                 methods = [ssh_transfer_method]
-        else:
-            # since this can be a non-bool now, we need to handle it correctly
-            if not isinstance(scp_if_ssh, bool):
-                scp_if_ssh = scp_if_ssh.lower()
-                if scp_if_ssh in BOOLEANS:
-                    scp_if_ssh = boolean(scp_if_ssh, strict=False)
-                elif scp_if_ssh != 'smart':
-                    raise AnsibleOptionsError('scp_if_ssh needs to be one of [smart|True|False]')
-            if scp_if_ssh == 'smart':
-                methods = smart_methods
-            elif scp_if_ssh is True:
-                methods = ['scp']
-            else:
-                methods = ['sftp']
 
         for method in methods:
```

Comment:
@Akasurde this line was previously fully covered but now the branch coverage is partial. This means that the tests don't cover the branch when the body of the for-loop is skipped, when `methods == []`, for example. Could you look into restoring coverage for this case?

Reply:
Consequently, line 1312/1277 through the end of the method are not covered at all (due to tests always hitting a early `return`).

Reply:
Finally, 1388/1353 now only has partial coverage (the tests only hit the else-branch, never raising the exception).